collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -126,6 +127,7 @@ public void removeTempInstance() {
                 FileUtils.purgeMediaPath(instanceFolder);
             }
         }
+        releaseMediaManager();
```

Comment:
Do we need to do this in `onCleared()` as well? Have you played around with the behaviour when "Don't save activities" is enabled? It might be good to give that a go and see if we need to handle the case when we leave the app, Android kills the process, and then we return to the app. Would files get left undeleted but on in the form in this case? Would things break?

Let me know if you have questions about trying that out!

Reply:
This is an interesting case. I will try that out with all the media widgets.

Reply:
Hi @seadowg I tried clearing the `originalFiles` and `recentFiles` HashMap in `onCleared()`, when `Don't keep activities` is turned on. I didn't find any anomaly in saving answers to the widgets, as I think, the answer is already saved with the widget, so clearing the files won't cause any break. Am I missing some test case here?

Reply:
Here's the problem I found:

1. Enabled "Don't keep activities"
1. Open the "All widgets form"
1. Go to the file widget
1. Select a file
1. Select a different file
1. Go to the hierarchy view with the arrow button
1. Press "Go To End"
1. Save and exit the form

If you now go to the "Device File Explorer" and check the instance dir (`sdcard/odk/instances` and then the newest dir) you'll see both the files you've selected are still in the directory. I'm guessing this is because when the activity is killed the view model state is wiped with the activity and the `originalFiles` list will be empty when we return to the `FormEntryActivity` from the hierarchy screen. Before (on `master`) this worked because the `MediaManager` was a process level singleton and so state would be retained. 

It feels to me like it would be a lot nicer if when the form was saved or (or ignore changes is hit) it just deleted any files not being used in the form but that could require a lot of rework. I'm thinking we should use the new [Saved State module](https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate) to persist `originalFiles` between activity instances (as opposed to going back to a process singleton) and then file an issue around carrying out the rework.

I'd like to pull @lognaturel and @grzesiek2010 for sanity checks here though!

Reply:
This case does seem possible if someone needs to get clarification on what file to attach, for example. Certainly it'd be best not to have a known regression in behavior.

We haven't used the saved state module yet. It might be simpler to use `onSavedInstanceState`. But yes, saving `originalFiles` between activity instances for now seems good.